app/models/katello/glue/candlepin/pool.rb

```diff
@@ -6,179 +6,134 @@ def self.included(base)
-          case attr['name']
-          when 'virt_limit'
-            @virt_limit = attr['value'].to_i
-          when 'support_type'
-            @support_type = attr['value']
-          when 'arch'
-            @arch = attr['value']
-          when 'support_level'
-            @support_level = attr['value']
-          when 'sockets'
-            @sockets = attr['value'].to_i
-          when 'cores'
-            @cores = attr['value'].to_i
-          when 'ram'
-            @ram = attr['value'].to_i
-          when 'description'
-            @description = attr['value']
-          when 'product_family'
-            @product_family = attr['value']
-          when 'variant'
-            @variant = attr['value']
-          when 'multi-entitlement'
-            @multi_entitlement = (attr['value'] == 'true' || attr['value'] == 'yes') ? true : false
-          when 'stacking_id'
-            @stacking_id = attr['value']
-          when 'instance_multiplier'
-            @instance_multiplier = attr['value'].to_i
-          end
-        end if attrs['productAttributes']
+        pool_attributes[:host_id] = pool_attributes["requiresHost"] if pool_attributes.key?("requiresHost")
 
-        @suggested_quantity = 1
-        attrs['calculatedAttributes'].each_key do |key|
-          case key
-          when 'suggested_quantity'
-            @suggested_quantity = attrs['calculatedAttributes']['suggested_quantity'].to_i
-          end
-        end if attrs['calculatedAttributes']
+        exceptions = pool_attributes.keys.map(&:to_sym) - self.attribute_names.map(&:to_sym)
+        self.update_attributes(pool_attributes.except!(*exceptions))
+        self.save!
+        self.create_activation_key_associations
       end
 
       def products
-        Katello::Product.where(:cp_id => provided_products.map { |prod| prod[:productId] })
+        self.subscription.products if self.subscription
       end
 
       def systems
         System.all_by_pool(cp_id)
       end
 
-      def activation_keys
+      def create_activation_key_associations
         keys = Resources::Candlepin::ActivationKey.get(nil, "?include=id&include=pools.pool.id")
         activation_key_ids = keys.collect do |key|
           key['id'] if key['pools'].any? { |pool| pool['pool']['id'] == cp_id }
         end
+        related_keys = ::Katello::ActivationKey.where(:cp_id => activation_key_ids.compact)
+        related_keys.each do |key|
+          Katello::PoolActivationKey.find_or_create_by_activation_key_id_and_pool_id(key.id, self.id)
+        end
+      end
 
-        return Katello::ActivationKey.where(:cp_id => activation_key_ids.compact)
+      def distributors
+        Distributor.all_by_pool(cp_id)
       end
 
       def host
-        System.find_by_uuid(host_id) if host_id
+        System.find_by_uuid(self.host_id) if self.host_id
       end
 
-      attr_reader :amount
+      def quantity_available
+        self.quantity - self.consumed if self.quantity && self.consumed
+      end
```

Comment:
quantity and consumed is in the AR record right?  If so i'd move this method to the main Pool  class